core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/reconfiguration/RestartStrategy.java

```diff
@@ -0,0 +1,38 @@
+package ai.timefold.solver.core.impl.localsearch.decider.reconfiguration;
+
+import ai.timefold.solver.core.impl.localsearch.decider.acceptor.restart.StuckCriterion;
+import ai.timefold.solver.core.impl.phase.event.PhaseLifecycleListener;
+import ai.timefold.solver.core.impl.phase.scope.AbstractStepScope;
+
+/**
+ * Base contract for defining restart strategies.
+ * The restart process is initiated
+ * when the {@link ai.timefold.solver.core.impl.localsearch.decider.LocalSearchDecider decider} identifies
```

Comment:
Where possible, please use imports. FQNs arguably don't provide much benefit, but they hinder readability of those comments.